parser/worker.go

```diff
@@ -262,44 +262,67 @@ func (w Worker) ExportCommit(commit *tmtypes.Commit, vals *tmctypes.ResultValida
 	return nil
 }
 
-// ExportTxs accepts a slice of transactions and persists then inside the database.
+// SaveTxs accepts the transaction and persists it inside the database.
 // An error is returned if the write fails.
-func (w Worker) ExportTxs(txs []*types.Tx) error {
-	// Handle all the transactions inside the block
-	for _, tx := range txs {
-		// Save the transaction itself
-		err := w.db.SaveTx(tx)
+func (w Worker) SaveTxs(tx *types.Tx) error {
+	err := w.db.SaveTx(tx)
+	if err != nil {
+		return fmt.Errorf("failed to handle transaction with hash %s: %s", tx.TxHash, err)
+	}
+	return nil
+}
+
+// HandleTxs accepts the transaction and calls the tx handlers.
+func (w Worker) HandleTxs(tx *types.Tx) {
```

Comment:
I think it's better to rename this `handleTx` (since it handles only a single transaction) making it private within the `Worker` type (instead of public) as well